src/lib.rs

```diff
@@ -342,43 +342,57 @@ impl PrometheusMetricsBuilder {
         self
     }
 
-    /// Report HTTP version of served request as "version" label
-    pub fn enable_http_version_label(mut self, enable: bool) -> Self {
-        self.enable_http_version_label = enable;
+    /// Set metrics configuration
+    pub fn metrics_configuration(mut self, value: ActixMetricsConfiguration) -> Self {
+        self.metrics_configuration = value;
         self
     }
 
     /// Instantiate PrometheusMetrics struct
     pub fn build(self) -> Result<PrometheusMetrics, Box<dyn std::error::Error + Send + Sync>> {
-        let http_requests_total_opts =
-            Opts::new("http_requests_total", "Total number of HTTP requests")
-                .namespace(&self.namespace)
-                .const_labels(self.const_labels.clone());
 
-        let label_names = if self.enable_http_version_label {
-            ["version", "endpoint", "method", "status"].as_slice()
-        } else {
-            ["endpoint", "method", "status"].as_slice()
-        };
+        let labels_vec = self.metrics_configuration.labels.clone().to_vec();
+        let labels = &labels_vec.iter().map(|s| s.as_str()).collect::<Vec<&str>>();
+
+        let http_requests_total_opts = Opts::new(
+            self.metrics_configuration
+                .http_requests_total_name
+                .to_owned(),
+            "Total number of HTTP requests",
+        )
+        .namespace(&self.namespace)
+        .const_labels(self.const_labels.clone());
 
-        let http_requests_total = IntCounterVec::new(http_requests_total_opts, label_names)?;
+        let http_requests_total = IntCounterVec::new(
+            http_requests_total_opts,
+            labels
+        )?;
 
         let http_requests_duration_seconds_opts = HistogramOpts::new(
-            "http_requests_duration_seconds",
+            self.metrics_configuration
+                .http_requests_duration_seconds_name
+                .to_owned(),
             "HTTP request duration in seconds for all requests",
         )
         .namespace(&self.namespace)
         .buckets(self.buckets.to_vec())
         .const_labels(self.const_labels.clone());
 
-        let http_requests_duration_seconds =
-            HistogramVec::new(http_requests_duration_seconds_opts, label_names)?;
+        let http_requests_duration_seconds = HistogramVec::new(
+            http_requests_duration_seconds_opts,
+            labels
+        )?;
 
         self.registry
             .register(Box::new(http_requests_total.clone()))?;
         self.registry
             .register(Box::new(http_requests_duration_seconds.clone()))?;
 
+        let enabled_http_version_label = match self.metrics_configuration.labels.version {
```

Comment:
Is it possible to use `self.metrics_configuration.labels.version.is_some`?

Reply:
Yes, of course. Thanks!